TYPO3.Flow/Classes/TYPO3/Flow/Mvc/ActionRequest.php

```diff
@@ -469,11 +469,12 @@ public function getControllerActionName()
      */
     public function setArgument($argumentName, $value)
     {
-        if (!is_string($argumentName) || strlen($argumentName) === 0) {
+        $argumentName = (string)$argumentName;
+        if (empty($argumentName)) {
             throw new Exception\InvalidArgumentNameException('Invalid argument name (must be a non-empty string).', 1210858767);
         }
 
-        if (substr($argumentName, 0, 2) === '__') {
+        if (strpos($argumentName, '__') === 0) {
```

Comment:
Yep, according to http://maettig.com/code/php/php-performance-benchmarks.php#startswith `strpos` is faster, though `strncmp($argumentName, '__', 2)` would be even faster then.

I wonder how `strpos` performs in a non-matching case though, because obviously it would need to go through the whole input string then.